src/pydvl/influence/base_influence_model.py

```diff
@@ -0,0 +1,173 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Collection, Generic, Iterable, Optional, TypeVar
+
+
+class InfluenceType(str, Enum):
+    r"""
+    Enum representation for the types of influence.
+
+    Attributes:
+        Up: Up-weighting a training point, see section 2.1 of
+            (Koh and Liang, 2017)<sup><a href="#koh_liang_2017">1</a></sup>
+        Perturbation: Perturb a training point, see section 2.2 of
+            (Koh and Liang, 2017)<sup><a href="#koh_liang_2017">1</a></sup>
+
+    """
+
+    Up = "up"
+    Perturbation = "perturbation"
+
+
+class UnsupportedInfluenceTypeException(ValueError):
+    def __init__(self, influence_type: str):
+        super().__init__(
+            f"Provided {influence_type=} is not supported. Choose one of InfluenceType.Up "
+            f"and InfluenceType.Perturbation"
+        )
+
+
+class NotFittedException(ValueError):
+    def __init__(self):
+        super().__init__(
+            f"Objects of type InfluenceFunctionModel must be fitted before calling influence methods. "
+            f"Call method fit with appropriate input."
+        )
+
+
+"""Type variable for tensors, i.e. sequences of numbers"""
+TensorType = TypeVar("TensorType", bound=Collection)
+DataLoaderType = TypeVar("DataLoaderType", bound=Iterable)
+
+
+class InfluenceFunctionModel(Generic[TensorType, DataLoaderType], ABC):
+    """
+    Generic abstract base class for computing influence related quantities. For a specific influence algorithm and
+    tensor framework, inherit from this base class
+    """
+
+    @property
+    @abstractmethod
+    def n_parameters(self):
+        """Number of trainable parameters of the underlying model"""
+
+    @property
+    @abstractmethod
+    def is_fitted(self):
+        """Overwrite this, to expose the fitting status of the instance."""
+
+    @abstractmethod
+    def fit(self, data_loader: DataLoaderType):
+        """
+        Overwrite this method to fit the influence function model to training data, e.g. pre-compute hessian matrix
```

Comment:
"Override" (and elsewhere too). Missing preposition and article
```suggestion
        Override this method to fit the influence function model to training data, e.g. to pre-compute the hessian matrix
```
Also, please format to 88 cols like the rest of the code

Reply:
Yes, override is correct....
This means black is not working for the docstrings?